### PR TITLE
Write exception handling for application, CRUD for entities, unit tests for services…

### PR DESCRIPTION
**Pull request by GoharHakobyan95**
…ts for service.

### CODE REVIEW

File: src/main/java/am/itspace/sweetbakerystorerest/config/WebSecurityConfig.java [hunk 1]
```diff
@@ -33,6 +37,15 @@ protected void configure(HttpSecurity http) throws Exception {
                 .exceptionHandling().authenticationEntryPoint(unauthorizedHandler)
                 .and()
                 .authorizeRequests()
+                .antMatchers(HttpMethod.GET, "/api/city").hasAuthority(Role.USER.name())
+                .antMatchers( "/api/city/**").hasAuthority(Role.ADMIN.name())
+                .antMatchers(HttpMethod.GET, "/api/payment").hasAuthority(Role.ADMIN.name())
```
Comment (Vahe650) [inline src/main/java/am/itspace/sweetbakerystorerest/config/WebSecurityConfig.java:41]: hasAnyAutority(user,admin .....)

File: src/main/java/am/itspace/sweetbakerystorerest/endpoint/CityEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+@RequiredArgsConstructor
+@Slf4j
+public class CityEndpoint {
+
+    private final CityService cityService;
+    private final CityMapper cityMapper;
+
+    @GetMapping
+    public ResponseEntity<List<City>> getAllCities(@AuthenticationPrincipal CurrentUser currentUser) {
+        log.info("Endpoint cities called by {}", currentUser.getUser().getEmail());
```
Comment (Vahe650) [inline src/main/java/am/itspace/sweetbakerystorerest/endpoint/CityEndpoint.java:41]: Page<City>

File: src/main/java/am/itspace/sweetbakerystorerest/endpoint/CityEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+        }
+        return ResponseEntity.ok(cityMapper.map(cityById.get()));
+    }
+
+
+    @PostMapping
+    private ResponseEntity<City> saveCity(@AuthenticationPrincipal CurrentUser currentUser,
+                                       @RequestBody @Valid CreateCityDto createCityDto) {
+        log.info("City {} has been saved by Admin ", currentUser.getUser().getEmail());
```
Comment (Vahe650) [inline src/main/java/am/itspace/sweetbakerystorerest/endpoint/CityEndpoint.java]: Loging must be after saving ,not before

File: src/main/java/am/itspace/sweetbakerystorerest/endpoint/OrderEndpoint.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+@Slf4j
+public class OrderEndpoint {
+    private final OrderService orderService;
+    private final ProductService productService;
+
+    private final OrderMapper orderMapper;
+
+    // Endpoint for user to show his all orders
+    @GetMapping
+    public ResponseEntity<List<OrderResponseDto>> getAllOrders(@AuthenticationPrincipal CurrentUser currentUser) {
```
Comment (Vahe650) [inline src/main/java/am/itspace/sweetbakerystorerest/endpoint/OrderEndpoint.java]: Page

File: src/main/java/am/itspace/sweetbakerystorerest/endpoint/PaymentEndpoint.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+@Slf4j
+public class PaymentEndpoint {
+
+    private final PaymentService paymentService;
+    private final PaymentMapper paymentMapper;
+
+    //   Admin can get all user's payments
+    @GetMapping
+    public ResponseEntity<List<?>> getAllPayments(@AuthenticationPrincipal CurrentUser currentUser) {
+        log.info("Endpoint payments called by {}", currentUser.getUser().getEmail());
```
Comment (Vahe650) [inline src/main/java/am/itspace/sweetbakerystorerest/endpoint/PaymentEndpoint.java:39]: Page<Payment>

**Review by GoharHakobyan95 — COMMENTED** [review]
Changes are made.

**Pull request merged**